handlers.go

```diff
@@ -11,11 +11,14 @@ import (
 	"io/ioutil"
 	"log"
 	"net/http"
+	"sync"
 )
 
 // A autographer signs input data with a private key
 type autographer struct {
-	signers []Signer
+	signers    []Signer
+	nonces     []nonce
+	noncesLock sync.Mutex
```

Comment:
why not: 

````
type autographer struct {
   sync.Mutex
   signers []Signer
}
````

Then you can: 

func (a *autographer) SomeFunc() {
   a.Lock()
   defer a.Unlock()
}

It makes cleaner looking code.